miner/worker.go

```diff
@@ -424,23 +417,29 @@ func (self *worker) commitNewWork() {
 		tstamp = parent.Time().Int64() + 1
 	}
 	// this will ensure we're not going off too far in the future
-	if now := time.Now().Unix(); tstamp > now+4 {
+	if now := time.Now().Unix(); tstamp > now+1 {
```

Comment:
@holiman Btw, your question here still stands. I'm not sure why this original 4 second into-the-future thing was allowed in the first place. My "fixed" code just ensures that future blocks cannot be minted. 

I'm guessing it was meant for the Frontier launch (or testnets) when difficulty was low and miners found multiple blocks a second. However I think it's an ugly and unnecessary hack really to allow future blocks.